File: HackerRank/30DaysOfCode/day2.php
<?php

// Complete the solve function below.
function solve($meal_cost, $tip_percent, $tax_percent)
{
    $total_percent = 100 + $tip_percent + $tax_percent;
    $total_cost = round($meal_cost * $total_percent / 100);

    print ($total_cost) . PHP_EOL;
}

$stdin = fopen("php://stdin", "r");

fscanf($stdin, "%lf\n", $meal_cost);

fscanf($stdin, "%d\n", $tip_percent);

fscanf($stdin, "%d\n", $tax_percent);

solve($meal_cost, $tip_percent, $tax_percent);

fclose($stdin);

File: HackerRank/30_days_of_code/day2/day2.rb
#!/bin/ruby

require "json"
require "stringio"

# Complete the solve function below.
def solve(meal_cost, tip_percent, tax_percent)
  tip_cost = meal_cost * (tip_percent / 100.0)
  tax_cost = meal_cost * (tax_percent / 100.0)
  return (meal_cost + tip_cost + tax_cost).round
end

#meal_cost = gets.to_f
#tip_percent = gets.to_i
#tax_percent = gets.to_i
#solve meal_cost, tip_percent, tax_percent

require "minitest/autorun"

class HackerRankTest < MiniTest::Test
  def test_solve
    assert_equal(15, solve(12.00, 20, 8))
    assert_equal(19, solve(15.50, 15, 10))
  end
end
